packages/ui/src/views/repo-list/repo-list-page.tsx

```diff
@@ -434,6 +435,23 @@ const SandboxRepoListPage: React.FC<RepoListProps> = ({ repositories, totalPages
     }).replace('about ', '')
   }))
 
+  const { query, handleSearch } = useCommonFilter()
+  const [value, setValue] = useState<string>()
+
+  useEffect(() => {
+    setValue(query || '')
+  }, [query])
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e?.target?.value)
+    handleSearch(e)
+  }
+
+  const handleResetQuery = () => {
+    setValue('')
+    handleSearch({ target: { value: '' } } as React.ChangeEvent<HTMLInputElement>)
```

Comment:
NIT: we can create a function within `useCommonFilter` hook to reset the filter.